universal-login-react/src/ui/commons/GasPrice/GasPrice.tsx

```diff
@@ -115,13 +120,32 @@ export const GasPrice = ({isDeployed = true, deployedWallet, sdk, gasLimit, onGa
       </div>
     </>
   );
+
   return (
     <div ref={ref} className={useClassFor('gas-price')}>
       {gasModes ? renderComponent(gasModes) : <Spinner className="spinner-small" />}
     </div>
   );
 };
 
+export const GasPrice = ({isDeployed = true, deployedWallet, sdk, gasLimit, onGasParametersChanged, className}: GasPriceProps) => {
+  const classNameForGasPrice = useClassFor('gas-price');
+  useEffect(() => {
+    if (sdk.isRefundPaid()) {
+      onGasParametersChanged(GAS_PRICE_FOR_NO_REFUND);
+    }
+  });
+
+  return sdk.isRefundPaid() ? <div className={classNameForGasPrice}></div> : <GasPriceWithOptions
```

Comment:
Can we inline `classNameForGasPrice`?

Reply:
Ok, in the next pr we will extract something like information `This transaction is refunded by ApplicationX`